pkg/aws/actuator/actuator.go

```diff
@@ -134,9 +135,9 @@ func (a *AWSActuator) Exists(ctx context.Context, cr *minterv1.CredentialsReques
 
 }
 
-// NeedsUpdate will return whether the current credentials satisfy what's being requested
+// needsUpdate will return whether the current credentials satisfy what's being requested
 // in the CredentialsRequest
-func (a *AWSActuator) NeedsUpdate(ctx context.Context, cr *minterv1.CredentialsRequest) (bool, error) {
+func (a *AWSActuator) needsUpdate(ctx context.Context, cr *minterv1.CredentialsRequest) (bool, error) {
```

Comment:
What purpose does needsUpdate serve? It feels like we do a lot of the same style of checking both here and in sync. Would it not make sense for sync to just go piece by piece and check if that part is ok, if not fix it?

Reply:
as discussed offline, needsUpdate() can determine whether we need an update in the case where the root creds are missing.

but if we need an update, we need to know whether to mint or passthrough which requires the existence of the root creds.

so needsUpdate() lets us succeed in the case where root creds are missing, but there is no need to update.

Reply:
Got it, thanks!